### DISCUSSION TITLE
Render image from markdown

### DISCUSSION THREAD

**Discussion by SebastianBienert** [scalar/scalar#842] [Q&A] [answered]
It seems like rendering images from markdown is currently not possible, all img tags are filtered out in `remark` configuration:

https://github.com/scalar/scalar/blob/d9b6d8104dd0f1997e45a5220c8bbc11ae2062c4/packages/api-reference/src/components/Content/MarkdownRenderer.vue#L25C1-L30C9

Is there any reason for doing it? I couldn't find any information in the git history, looks like this filtering was set up there from the beginning.  Could it be removed?

**Comment by hanspagel** [comment]
Yeah, external requests/image rendering can cause a lot of trouble:

* You could use them for tracking purposes
* Loading huge images would block the network (or waste mobile bandwidth)
* Often breaks the layout and never really looks good

What’s the use case for images in the OpenAPI file? I’m open to reconsider the decision.

**Comment by SebastianBienert** [comment]
It is related to my other [question](https://github.com/scalar/scalar/discussions/781). I want to expand documentation with some static data, like overall architecture diagrams, description of use cases etc. From what I understood I could do it by adding this content as a markdown in  OpenAPI spec file. All headers would be then reflected in the sidebar and navigation/search would work the same way as it were a regular endpoint documentation. If there is another way to extend sidebar with static content, please let me know

**Reply by hanspagel** [reply, marked as answer]
Got it, sounds legit. We’ll add support for images. :)

**Comment by arbh89** [comment]
I was struggling with this too, i was trying to understand why images were not rendered and found the same as @SebastianBienert, in my case i am trying to include images for better documentation using ilustritions of API flows.
